developer_guides/topology2/topology2.rst

```diff
@@ -0,0 +1,895 @@
+.. _topology2:
+
+Topology 2.0
+############
+
+This is a high-level keyword extension on top of the existing ALSA conf topology format designed
+to:
+
+* Simplify the ALSA conf topology definitions by providing high level "classes" so topology
+  designers need to write less config for commonly defined objects.
+
+* Allow simple reuse of objects. Define once and reuse (like M4) with the ability to alter objects
+  configuration attributes from defaults.
+
+* Allow data type and value verification. This is not done today and frequently crops up in FW bug
+  reports.
+
+.. contents::
+
+1. Ingredients
+**************
+
+A typical 2.0 configuration file consists of the following:
+
+1. Classes
+2. Objects
+3. Arguments
+
+1.1 Classes
+***********
+
+Topology today has some common definitions that are often reused throughout with slightly altered
+configurations such as widgets (components), pipelines, dais, pcm and controls. Topology 2.0
+introduces the concept of reusable "class" like definitions that can be used to create commonly
+used topology objects. Classes are defined with a new keyword “Class”.
+
+A class definition always starts withe "Class" keyword followed by 2 nodes. The first node contains
+the class group and the second node contains the class name. For example:
+
+.. code-block:: bash
+
+	Class.Base.data {}
+
+Note that '.' is the node separator in the alsaconf syntax. In the above line, "Base" is the class
+group and "data" is the class name.
```

Comment:
are the class groups to be selected from a predefined list or can they be chosen freely?